Compiler warnings

I see several compiler warnings when doing a standard build on an Ubuntu 14.04 box.

E.g.

```
kexhyecdhoqsc.c:145:14: warning: variable ‘kex’ set but not used [-Wunused-but-set-variable]
  struct kex *kex = NULL;
              ^
kexpqoqsc.c:196:14: warning: variable ‘kex’ set but not used [-Wunused-but-set-variable]
  struct kex *kex = NULL;
              ^
kexpqoqsc.c: In function ‘pq_oqs_client’:
kexoqs.c:286:1: warning: label ‘out’ defined but not used [-Wunused-label]
 out:
 ^
kexhyecdhoqsc.c: In function ‘input_hybrid_ecdh_oqs_reply’:
kexhyecdhoqsc.c:328:3: warning: implicit declaration of function ‘memcpy’ [-Wimplicit-function-declaration]
   memcpy(kex->session_id, hash, kex->session_id_len);
   ^
kexhyecdhoqsc.c:328:3: warning: incompatible implicit declaration of built-in function ‘memcpy’ [enabled by default]
kexpqoqsc.c:276:3: warning: passing argument 1 of ‘pq_oqs_free’ from incompatible pointer type [enabled by default]
   pq_oqs_free(oqs_kex_ctx);
```

Should be quick to solve by adding missing includes and remove unused variables.

--- a/kexhyecdhoqs.c
+++ b/kexhyecdhoqs.c
@@ -26,6 +26,8 @@
 
 #if defined(OPENSSL_HAS_ECC) && defined(WITH_OQS) && defined(WITH_HYBRID_KEX)
 
+#include <string.h>
+
 #include "ssherr.h"
 #include "digest.h"
 #include "ssh2.h"

--- a/kexhyecdhoqsc.c
+++ b/kexhyecdhoqsc.c
@@ -28,6 +28,7 @@
 
 #include <openssl/ecdh.h>
 #include <signal.h>
+#include <string.h>
 
 #include "sshkey.h"
 #include "digest.h"
@@ -142,13 +143,10 @@ hybrid_ecdh_oqs_deserialise_hostkey(struct ssh *ssh,
 	struct sshkey **server_host_key, u_char **server_host_key_blob,
 	size_t *server_host_key_blob_len) {
 
-	struct kex *kex = NULL;
 	struct sshkey *tmp_server_host_key = NULL;
 	u_char *tmp_server_host_key_blob = NULL;
 	size_t tmp_server_host_key_blob_len = 0;
 	int r = 0;
-
-	kex = ssh->kex;
 
 	/* Extract host key from packet */
 	if ((r = sshpkt_get_string(ssh, &tmp_server_host_key_blob,

--- a/kexhyecdhoqss.c
+++ b/kexhyecdhoqss.c
@@ -28,6 +28,7 @@
 
 #include <openssl/ecdh.h>
 #include <signal.h>
+#include <string.h>
 
 #include "sshkey.h"
 #include "digest.h"
@@ -37,6 +38,7 @@
 #include "dispatch.h"
 #include "packet.h"
 #include "sshbuf.h"
+#include "log.h"
 
 /* Server private */
 static int

--- a/kexoqs.c
+++ b/kexoqs.c
@@ -26,6 +26,8 @@
 
 #ifdef WITH_OQS
 
+#include <string.h>
+
 #include "ssherr.h"
 #include "packet.h"
 #include "ssh2.h"
@@ -262,13 +264,8 @@ int
 oqs_deserialise(struct ssh *ssh, OQS_KEX_CTX *oqs_kex_ctx,
 	enum oqs_client_or_server client_or_server) {
 
-	int r = 0;
-
-	r = sshpkt_get_string(ssh, &(oqs_kex_ctx->oqs_remote_msg),
+	return sshpkt_get_string(ssh, &(oqs_kex_ctx->oqs_remote_msg),
 		&(oqs_kex_ctx->oqs_remote_msg_len));
-
-out:
-	return r;
 }
 
 /*
@@ -278,13 +275,8 @@ int
 oqs_serialise(struct ssh *ssh, OQS_KEX_CTX *oqs_kex_ctx,
 	enum oqs_client_or_server client_or_server) {
 
-	int r = 0;
-
-	r = sshpkt_put_string(ssh, oqs_kex_ctx->oqs_local_msg,
+	return sshpkt_put_string(ssh, oqs_kex_ctx->oqs_local_msg,
 		oqs_kex_ctx->oqs_local_msg_len);
-
-out:
-	return r;
 }
 
 /*

--- a/kexpq.h
+++ b/kexpq.h
@@ -65,7 +65,7 @@ int pq_oqs_hash (
 	u_char *hash, size_t *hash_len);
 /* Shared functions */
 int pq_oqs_init(PQ_KEX_CTX **pq_kex_ctx, char *pq_kex_name);
-int pq_oqs_free(PQ_KEX_CTX *pq_kex_ctx);
+void pq_oqs_free(PQ_KEX_CTX *pq_kex_ctx);
 /* Client specific function */
 int  pq_oqs_client(struct ssh *);
 /* Server specific function */

--- a/kexpqoqs.c
+++ b/kexpqoqs.c
@@ -93,7 +93,7 @@ out:
 /*
  * @brief Free memory allocated PQ-only key exchange liboqs
  */
-int
+void
 pq_oqs_free(PQ_KEX_CTX *pq_kex_ctx) {
 
 	if (pq_kex_ctx->oqs_kex_ctx != NULL) {

--- a/kexpqoqsc.c
+++ b/kexpqoqsc.c
@@ -27,6 +27,7 @@
 #if defined(WITH_OQS) && defined(WITH_PQ_KEX)
 
 #include <signal.h>
+#include <string.h>
 
 #include "sshkey.h"
 #include "digest.h"
@@ -92,64 +93,6 @@ pq_oqs_c2s_serialise(struct ssh *ssh, PQ_KEX_CTX *pq_kex_ctx) {
 }
 
 /*
- * @brief Extracts host key from incoming packet and
- * verifies it
- */
-static int
-pq_oqs_server_hostkey(struct ssh *ssh,
-	struct sshkey **server_host_key, u_char **server_host_key_blob,
-	size_t *server_host_key_blob_len) {
-
-	struct kex *kex = NULL;
-	struct sshkey *tmp_server_host_key = NULL;
-	u_char *tmp_server_host_key_blob = NULL;
-	size_t tmp_server_host_key_blob_len = 0;
-	int r = 0;
-
-	kex = ssh->kex;
-
-	/* If we can't verify the host key then abort */
-	if (kex->verify_host_key == NULL) {
-		r = SSH_ERR_INVALID_ARGUMENT;
-		goto out;
-	}
-
-	/* Extract host key from packet */
-	if ((r = sshpkt_get_string(ssh, &tmp_server_host_key_blob,
-		&tmp_server_host_key_blob_len)) != 0 ||
-		(r = sshkey_from_blob(tmp_server_host_key_blob,
-		tmp_server_host_key_blob_len, &tmp_server_host_key)) != 0)
-		goto out;
-	if (tmp_server_host_key->type != kex->hostkey_type ||
-	    (kex->hostkey_type == KEY_ECDSA &&
-	    tmp_server_host_key->ecdsa_nid != kex->hostkey_nid)) {
-		r = SSH_ERR_KEY_TYPE_MISMATCH;
-		goto out;
-	}
-
-	/* Verify host key */
-	if (kex->verify_host_key(tmp_server_host_key, ssh) == -1) {
-		r = SSH_ERR_SIGNATURE_INVALID;
-		goto out;
-	}
-
-	*server_host_key = tmp_server_host_key;
-	*server_host_key_blob = tmp_server_host_key_blob;
-	*server_host_key_blob_len = tmp_server_host_key_blob_len;
-
-	tmp_server_host_key = NULL;
-	tmp_server_host_key_blob = NULL;
-
-out:
-	if (tmp_server_host_key_blob != NULL)
-		free(tmp_server_host_key_blob);
-	if (tmp_server_host_key != NULL)
-		sshkey_free(tmp_server_host_key);
-
-	return r;
-}
-
-/*
  * @brief Verifies host key
  */
 static int
@@ -193,13 +136,10 @@ pq_oqs_deserialise_hostkey(struct ssh *ssh,
 	struct sshkey **server_host_key, u_char **server_host_key_blob,
 	size_t *server_host_key_blob_len) {
 
-	struct kex *kex = NULL;
 	struct sshkey *tmp_server_host_key = NULL;
 	u_char *tmp_server_host_key_blob = NULL;
 	size_t tmp_server_host_key_blob_len = 0;
 	int r = 0;
-
-	kex = ssh->kex;
 
 	/* Extract host key from packet */
 	if ((r = sshpkt_get_string(ssh, &tmp_server_host_key_blob,
@@ -273,7 +213,7 @@ pq_oqs_client(struct ssh *ssh) {
 
 out:
 	if (r != 0)
-		pq_oqs_free(oqs_kex_ctx);
+		pq_oqs_free(pq_kex_ctx);
 
 	return r;
 }

--- a/kexpqoqss.c
+++ b/kexpqoqss.c
@@ -27,6 +27,7 @@
 #if defined(WITH_OQS) && defined(WITH_PQ_KEX)
 
 #include <signal.h>
+#include <string.h>
 
 #include "sshkey.h"
 #include "digest.h"

